src/tickit/core/components/system_simulation.py

```diff
@@ -65,8 +72,33 @@ async def on_tick(self, time: SimTime, changes: Changes) -> None:
             changes (Changes): A mapping of changed component inputs and their new
                 values.
         """
-        output_changes, call_in = await self.scheduler.on_tick(time, changes)
-        await self.output(time, output_changes, call_in)
+        on_tick_task = asyncio.create_task(self.scheduler.on_tick(time, changes))
+        error_state = asyncio.create_task(self.scheduler.error.wait())
+
+        done, _ = await asyncio.wait(
+            [on_tick_task, error_state],
+            return_when=asyncio.tasks.FIRST_COMPLETED,
+        )
+        if error_state in done:
+            await self.state_producer.produce(
+                output_topic(self.name),
+                self.scheduler.component_error,
+            )
+
+        elif on_tick_task in done:
+            output_changes, call_in = on_tick_task.result()
+            await self.output(time, output_changes, call_in)
+
+    async def stop_component(self) -> None:
+        """Cancel all pending tasks associated with the System Simulation component.
+
+        Cancels long running adapter tasks associated with the component.
+        """
+        LOGGER.debug("Stopping {}".format(self.name))
+        if self._tasks:
+            for task in self._tasks:
+                task.cancel()
+            await asyncio.wait(self._tasks)
```

Comment:
The `if` here is superfluous
```suggestion
        for task in self._tasks:
            task.cancel()
        await asyncio.wait(self._tasks)
```

Reply:
Not superfluous If, but moved where the if is as we are already awaiting on this somewhere else so we can remove this code and do the if there.

Reply:
Is it now just this?
```python
                for task in self._tasks:
	            task.cancel()
```
Making use of the pre-existing `asyncio.wait`

Reply:
Is it now just this?
```python
                for task in self._tasks:
	            task.cancel()
```
Making use of the pre-existing `asyncio.wait`